srttools/monitor/resources/index.html

```diff
@@ -0,0 +1,223 @@
+            let thumbnail = document.getElementById(`thumb-${String(index).padStart(3, '0')}`) || document.createElement("img");
+
+            if (base64 === "") {
+                thumbnail.remove();
+            } else {
+                thumbnail.classList.add("thumbnail");
+                thumbnail.id = `thumb-${String(index).padStart(3, '0')}`;
+                thumbnail.src = "data:image/png;base64," + base64;
+
+                let pair = getPair(Math.floor(index / 2));
+                pair.appendChild(thumbnail);
+                let children = Array.from(pair.children);
+                children.sort((a, b) => { return a.id.localeCompare(b.id); });
+                children.forEach(child => pair.appendChild(child));
+                bar.appendChild(pair);
+                children = Array.from(bar.children);
+                children.sort((a, b) => { return a.id.localeCompare(b.id); });
+                children.forEach(child => bar.appendChild(child));
+            }
+
+            updateMainView();
+        }
+
+        function connect() {
+            let ws = new WebSocket("ws://" + destination + ":$port/images");
+
+            ws.onmessage = function(message) {
+                let msg = JSON.parse(message.data);
+                requestAnimationFrame(() => {
+                    addImage(msg.index, msg.image);
+                });
+            };
+
+            ws.onclose = function() {
+                setTimeout(connect, $reconnectTime);
+            };
+        }
+
+        connect();
+
+        let wrapper = document.getElementById("thumbnail-wrapper");
+        wrapper.addEventListener("mouseleave", () => {
+            scrollSpeed = 0;
+        });
+
+        wrapper.addEventListener('mousemove', (event) => {
+            const clientX = event.clientX;
+            const screenWidth = window.innerWidth;
+            const threshold = Math.round(screenWidth * 0.2);
+
+            if (clientX < threshold) {
+                let distance = threshold - clientX;
+                let factor = Math.pow(distance / threshold, 3);
+                scrollSpeed = -factor * maxScrollSpeed;
+            } else if (clientX > screenWidth - threshold) {
+                let distance = clientX - (screenWidth - threshold - 1);
+                let factor = Math.pow(distance / threshold, 3);
+                scrollSpeed = factor * maxScrollSpeed;
+            } else {
+                scrollSpeed = 0;
+            }
+        });
+
+        function autoScroll() {
+            const barStyle = document.querySelector("#thumbnail-bar");
+            if (barWidth < window.innerWidth) {
+                bar.scrollLeft = 0;
+                barStyle.style.justifyContent = 'center';
+            } else {
+                barStyle.style.justifyContent = 'flex-start';
+                bar.scrollLeft += scrollSpeed;
+            }
+
+            requestAnimationFrame(autoScroll);
+        }
+
+        autoScroll();
+    </script>
+</body>
+</html>
```

Comment:
Moved the webpage code in a resource file by itself. It was too big to be embedded in python code cleanly.